### ISSUE TITLE
FUIFirestoreTableViewDataSource populateCell not called for document with modified field

### ISSUE THREAD

**Issue by alexroode** [firebase/FirebaseUI-iOS#420]
### Step 2: Describe your environment

  * Objective C or Swift: Swift 4
  * iOS version: 11.2
  * Firebase SDK version: 4.10.1
  * FirebaseUI version: 4.5.3
  * CocoaPods Version: 1.4
  
### Step 3: Describe the problem:

I am seeing that when using `FUIFirestoreTableViewDatasource`, if add to or remove documents (e.g. using the Firebase console) from a collection bound to a `UITableView`, the table immediately shows the added document or removes the deleted one. But if I modify a field of a document which is displayed in my tableView, `populateCell` is not called and the change is not reflected in the view (until the view is recreated or the cells are reused)

For example, with the following code, if I modify the `name` property of one of the documents in the `tests` collection, I still see the old name in my table. But if I add a new document, I immediately see it in the table.
```
let query = Firestore.firestore().collection("tests")
self.dataSource = FUIFirestoreTableViewDataSource(query: query, populateCell: { (tableView, indexPath, snapshot) -> UITableViewCell in
     let cell = tableView.dequeueReusableCell(withIdentifier: "cell")!
     let data = snapshot.data()!
     cell.textLabel?.text = data["name"] as? String
     return cell
  })
dataSource.bind(to: tableView)
  ```

If I add a listener using `query.addSnapshotListener`, the callback is called with the a new `QuerySnapshot` with the updated `name` immediately after I update its value, so I know the change is making it to the device successfully.

This is causing problems in my actual application - where I have a query sorted by an `order` property, and if I adjust the order in my back end, the displayed cell order on my device is now temporarily out of sync with the underlying data source. At that point I can tap on a cell and be redirected to my detail view for the wrong document.

This very similar code using the Realtime Database works as I expect. If I modify the `name` of a child of `tests`, I immediately see the new name in my `UITableView`.

```
let query = Database.database().reference(withPath: "tests")
self.dataSource = FUITableViewDataSource(query: query, populateCell: { (tableView, indexPath, snapshot) -> UITableViewCell in
    let cell = tableView.dequeueReusableCell(withIdentifier: "cell")!
    let data = snapshot.value as! [String: Any]
	cell.textLabel?.text = data["name"] as? String
	return cell
})
dataSource.bind(to: tableView)
```

**Comment by morganchen12** [comment]
I wasn't able to reproduce this. Here's my code:

```objc
- (void)viewDidAppear:(BOOL)animated {
  [super viewDidAppear:animated];

  self.user = self.auth.auth.currentUser;
  self.collection = [[FIRFirestore firestore] collectionWithPath:@"messages"];
  self.query = [self.collection queryOrderedByField:@"text"];

  _dataSource = [[FUICollectionViewDataSource alloc] initWithQuery:self.query
      populateCell:^UICollectionViewCell *(UICollectionView *collectionView,
                                           NSIndexPath *indexPath,
                                           FIRDocumentSnapshot *object) {
        FSCGlobalChatCollectionViewCell *cell =
            [collectionView dequeueReusableCellWithReuseIdentifier:@"GlobalChatCell"
                                                      forIndexPath:indexPath];
        FSCMessage *message = [[FSCMessage alloc] initWithDictionary:object.data];
        BOOL isSender = [message.userID isEqualToString:self.user.uid];
        [cell populateWithMessage:message isSender:isSender];
        NSLog(@"updated with message: %@", message);
        return cell;
  }];
  FSCGlobalChatLayout *layout = [[FSCGlobalChatLayout alloc] initWithDataSource:self.dataSource
                                                                         userID:self.user.uid];
  self.collectionView.collectionViewLayout = layout;

  [self.dataSource bindToView:self.collectionView];
}
```

Updating a document's fields in console causes an update on the client, which then refreshes correctly. Are there any errors in your console? If you're not running the latest version of Firestore, does updating make the issue go away?

**Comment by alexroode** [comment]
Thank you for looking into it.

I believe I am already running the latest version of everything:
* FirebaseCode 4.0.17
* FirebaseUI 4.5.3
* FirebaseFirestore 0.10.3

The only lines I am seeing in the console some log statements like these:
```
2018-03-13 21:57:25.884779-0700 FirestoreTest[89655:1770223] TIC Read Status [3:0x0]: 1:57
2018-03-13 21:57:26.300099-0700 FirestoreTest[89655:1770223] TIC Read Status [4:0x0]: 1:57
2018-03-13 21:57:26.300316-0700 FirestoreTest[89655:1770223] TIC Read Status [4:0x0]: 1:57
```
I have tried `FUICollectionViewDataSource`, moving the dataSource initialization and binding to `viewDidLoad`, and building the query using `collectionWithPath` and with `queryOrderedByField` as in your example - I see the same issue in all cases.

I have not tried adding auth - my test collection allows reads for all users.

I have uploaded a repository where I see the issue:
[https://github.com/alexroode/FirestoreTest](url)
With this code, when I update the "name" property of a document in the "tests" collection, I see the new name output to the console (in Xcode), but the old name is still visible in the tableView as populateCells has not been called.
